### PR TITLE
Add type to model and entity folder

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this breaks BC.

Helps on #1493 

## Changelog

```markdown
### Changed
- Adjust `PageInterface`, `SnapshotInterface` and `SiteInterface` to reflect all the methods available on the actual implementation.
```

### CODE REVIEW

**Review by VincentLanglet — COMMENTED** [review]
Why phpdoc over typehint ?

**Comment by jordisala1991** [comment]
> Why phpdoc over typehint ?

Because until I get phpstan up enough, I don't have enough confidence all of this is correct. Once the phpdoc is in place and correct, change it for actual typehint is easy.

**Comment by jordisala1991** [comment]
This is part of the PHPStan level 6 PR, but since it is growing a lot, I decided to split this folder in a separate PR.

File: tests/Model/PageTest.php [hunk 1]
```diff
@@ -163,12 +163,8 @@ public function testGetterSetter(): void
...
         static::assertSame($time, $page->getUpdatedAt());
 
-        $children = [
-            new Page(),
-            new Page(),
-        ];
-
-        $page->setChildren($children);
+        $page->addChild(new Page());
+        $page->addChild(new Page());
```
Comment (jordisala1991) [inline tests/Model/PageTest.php:167]: TBH I don't know why I cant do this:

```
        $children = new ArrayCollection([
            new Page(),
            new Page(),
        ]);

        $page->setChildren($children);
```

PHPStan complains:

```
  172    Parameter #1 $children of method Sonata\PageBundle\Model\Page::setChildren() expects
         Doctrine\Common\Collections\Collection<(int|string), Sonata\PageBundle\Model\PageInterface>,
         Doctrine\Common\Collections\ArrayCollection<int, Sonata\PageBundle\Tests\Model\Page> given.
```

wdyt @VincentLanglet , should we care about this? Do you know what is the problem?

**Comment by jordisala1991** [comment]
I know this one is hard to review...

What I did:
* Make sure interface and base class have the same methods.
* Make sure the properties on the base class are correct

To do so, I reordered the files in the same order, this way I saw some methods missing and other that were unused across the project.

File: tests/Model/PageTest.php (hunk 1 above)
Comment (VincentLanglet) [reply tests/Model/PageTest.php:167]: Weird since https://phpstan.org/r/ed0dcdc6-26c1-416c-8e99-012cfabd25f2

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**